finder_server.cc

```diff
@@ -115,8 +88,7 @@ vector<VendorInfo> SupplierClient::InquireVendorInfo (uint32_t food_id) {
   vector<VendorInfo> info;
   
   ClientContext context;
-  std::unique_ptr<ClientReader<VendorInfo>> reader (supplier_stub_->CheckVendor(&context, request));
-  // Status status = supplier_stub_->CheckVendor(&context, request, &info);
+  std::unique_ptr<ClientReader<VendorInfo>> reader(supplier_stub_->CheckVendor(&context, request));
 
   while (reader->Read(&reply)) {
     info.push_back(reply);
```

Comment:
Can you take advantage of streaming here too?

Reply:
I changed this part so that the finder server read only one vendor at a time and query that vendor. Although there might not be significant efficiency boost since the server need to save all the inventory information anyway, There are two potential benefits:
1. it balances the load of vendor server and the supplier server. 
2. it can be parallelized